pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -233,12 +236,35 @@ func (a *Azure) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretF
 	return secretsMap, nil
 }
 
+func (a *Azure) getSecretTag(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	version := ""
+	_, secretName, extraArg := getObjType(ref)
+
+	if ref.Version != "" {
+		version = ref.Version
+	}
+	
+	// returns a SecretBundle with the tag value
+	// https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/keyvault/v7.0/keyvault#SecretBundle
+	secretResp, err := a.baseClient.GetSecret(context.Background(), *a.provider.VaultURL, secretName, version)
+	if err != nil {
+		return nil, err
+	}
+	if extraArg == "" {
+		return nil, errors.New(errTagNotProvided)
+	}
```

Comment:
Hi @gusfcarvalho 
I'm working with the dragon on this PR
to me if a user wants to get the tag from a secret without specifying its name, it must be an error and no k8s secret created
if we return anything else, like the secret value, it will be too confusing
what do you think ?

Reply:
The idea would be to return all the tags of that secret, if no specific tag was provided. I'm not sure if this would cover any specific use case though. What I'm looking at is having the same experience in both ways (and this is also common in a lot of other providers).

Reply:
For the other providers, if we have an encoded information with `GetSecret`, we return the full raw value. In the case of the tags would be a JSON with all the tags.

Reply:
But retrieving all the tags is covered with datafrom,
I would find a bit confusing if you could do the same with data but just not providing an specific tag, don't you think?

Reply:
The main difference would be that one return would be a single kubernetes key, loaded into a JSON, while the other would be multiple K8s keys. So, for instance:
```
#metadata:
{
"yada":"yada",
"baz":"yey"
}
```
```
spec:
  data:
  - secretKey: foo
    remoteRef:
      key:tag/bar
  dataFrom:
  - extract:
      key: tag/bar
```
Should create a secret with 
```
foo: {"yada":"yada","baz":"yey"} # from secretKey + noRef
yada: yada # from dataFrom
baz: yey # from dataFrom
```

Again, there is no specific use case for it, but it is the general behaviour of `data` across providers that only support k/v method. So the idea here is just to make it a single experience for the developers (as far as we can).